internal/manager.go

```diff
@@ -30,10 +33,19 @@ func (m *CmdManager) RegisterCommand(cmd Command) error {
 	return nil
 }
 
+func (m *CmdManager) RegisterCommands(cmds []Command) error {
+	for _, cmd := range cmds {
+		if err := m.RegisterCommand(cmd); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 func (m *CmdManager) Run(task string, args ...string) error {
```

Comment:
I think you should change this code block to: 
* Add arg `ctx context.Context` as first argument of `CmdManager.Run` method
* Then pass ctx to the method `cmd.Execute(ctx,...)
```suggestion
func (m *CmdManager) Run(ctx context.Context, task string, args ...string) error {
        ...
	return cmd.Execute(ctx, args...)
}
```